### PR TITLE
Close #143241: Testing UI filter survives window reload

### LINKED ISSUE #143241: [Testing UI] Filter should survive 'Reload Window'
Issue Type: <b>Feature Request</b>
@connor4312 

The Test Explorer filter should be preserved if a window reload occurs, rather than resetting to default. This behavior could be gated by a setting.
image

VS Code version: Code - Insiders 1.65.0-insider (52e65499654de7525e30df3a8228b69320796bf5, 2022-02-16T05:16:18.653Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 4700U with Radeon Graphics          (8 x 1996)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.36GB (3.98GB free)|
|Process Argv|--folder-uri file:///c%3A/Users/JGrote/Projects/SeasonsHospiceDDL --crash-reporter-id 4453efaa-0821-4555-88e2-3de8bad9109e|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (32)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-color|ans|0.4.5
vscode-opennewinstance|chr|0.0.12
gitlens|eam|11.7.0
filter-lines|ear|1.0.0
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.2.0
copilot-nightly|Git|1.7.5030
remotehub|Git|0.24.0
gc-excelviewer|Gra|4.0.50
todo-tree|Gru|0.0.215
vscode-drawio|hed|1.6.4
change-case|hjd|1.0.2
vscode-peacock|joh|4.0.0
git-graph|mhu|1.30.0
vscode-azurefunctions|ms-|1.6.0
vscode-azureresourcegroups|ms-|0.4.0
vscode-docker|ms-|1.19.0
remote-containers|ms-|0.222.0
azure-account|ms-|0.10.0
powershell-preview|ms-|2022.2.1
indent-rainbow|ode|8.2.2
docthis|oou|0.8.2
pester-test|psp|2021.10.2
vscode-commons|red|0.0.6
vscode-yaml|red|1.4.0
vscode-inline-values-powershell|Tyl|0.0.5
errorlens|use|3.4.1
vscode-icons|vsc|11.10.0
codetour|vsl|0.0.58
gistfs|vsl|0.3.0
gitmoji-vscode|Vtr|1.0.9
better-align|wwm|1.1.6

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vsclayoutctrc:30405799
dsvsc008:30436272
pythonvsnew555:30438690
vsbas813:30436447
vscscmwlcmt:30428973
azactmsal:30432848
vscgsvid2:30433758
dockersurvey3:30438168

```

</details>

### PR DESCRIPTION
**Pull request by amanasifkhalid**
This PR fixes #143241. Upon loading the Testing UI, if the filter history contains a previous filter, it is loaded as the current value of the search bar, and only tests matching the filter are shown. Else, the search bar is empty, and all tests are shown. Below is a demo:

https://user-images.githubusercontent.com/58230338/163103028-e0c56d06-649e-40c3-9e0e-bfe0c6fdd2d7.mp4

### CODE REVIEW

File: src/vs/workbench/contrib/testing/browser/testingExplorerFilter.ts [hunk 1]
```diff
@@ -89,7 +95,7 @@ export class TestingExplorerFilter extends BaseActionViewItem {
 				value: this.state.text.value,
 				placeholderText: localize('testExplorerFilter', "Filter (e.g. text, !exclude, @tag)"),
 			},
-			history: this.history.get([])
+			history: history
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/browser/testingExplorerFilter.ts:98]:
```suggestion
			history
```

File: src/vs/workbench/contrib/testing/browser/testingExplorerFilter.ts [hunk 2]
```diff
@@ -65,6 +65,12 @@ export class TestingExplorerFilter extends BaseActionViewItem {
 		const wrapper = this.wrapper = dom.$('.testing-filter-wrapper');
 		container.appendChild(wrapper);
 
+		const history = this.history.get([]);
+		const previousValue = history ? history[history.length - 1] : null;
+		if (previousValue) {
+			this.state.setText(previousValue);
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/browser/testingExplorerFilter.ts:71]: `history` will always be defined, so

```suggestion
		if (history.length) {
			this.state.setText(history[history.length - 1]);
```

**Comment by connor4312** [comment]
Thanks for your contribution!

**Comment by amanasifkhalid** [comment]
Thank you for reviewing!

**Pull request merged**